src/components/slider/slider-utils.ts

```diff
@@ -1,30 +1,17 @@
 import type { DaikinSlider } from "./daikin-slider";
 
 /**
- * Change the slider properties from string to float for calculate.
+ * This function is used to handle floating-point errors in calculations.
+ * Ensuring that the number of decimal places matches the decimal places of the step value.
  *
- * @param min The min value of the slider.
- * @param max The max value of the slider.
- * @param step The step of the slider.
- * @param value The current value of the slider.
- * @returns An array of parsed float value.
+ * @param value The current value.
+ * @param step The step value.
+ * @returns Formatted current value.
  */
-export function parsedToFloat(
-  min: string,
-  max: string,
-  step: string,
-  value: string
-): number[] {
-  const minFloat = parseFloat(min);
-  const maxFloat = parseFloat(max);
-  const stepFloat = parseFloat(step);
-  const valueFloat = parseFloat(value);
-  return [
-    minFloat,
-    maxFloat,
-    stepFloat,
-    Math.max(minFloat, Math.min(valueFloat, maxFloat)),
-  ];
+export function formatValue(value: number, step: string) {
+  const decimals = step.includes(".") ? step.split(".")[1].length : 0;
+  const clampedValue = parseFloat(value.toFixed(decimals));
+  return `${clampedValue}`;
```

Comment:
Though it will work for most cases, converting number to string should be done using `.toFixed()`.

```suggestion
  // Get the length of fractional part excluding trailing zeros.
  const decimals = /\.(\d*?)0*$/.exec(step)?.[1].length ?? 0;
  // Format the value.
  const formattedValue = value.toFixed(decimals);
  // Remove trailing zeros if the fractional part exists.
  return decimals > 0 ? formattedValue.replace(/\.?0+$/, "") : formattedValue;
```